lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FilePartition.cs

```diff
@@ -0,0 +1,94 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Org.Apache.REEF.IO.FileSystem;
+using Org.Apache.REEF.IO.FileSystem.Hadoop;
+using Org.Apache.REEF.IO.PartitionedData.FileSystem.Parameters;
+using Org.Apache.REEF.IO.PartitionedData.Random.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IO.PartitionedData.FileSystem
+{
+    internal sealed class FilePartition<T> : IPartition<IEnumerable<T>>
```

Comment:
The `using` keyword calls Dispose. Registering Files for deletion is not exposed via a convenient .NET API, AFAIK. It can still be P/Invoked. I don't know if I would go that route.

Another solution here is to not use .NET File class directly, instead have a wrapper of TempFiles around .NET File type which deletes the file on Dispose.